subworkflows/nf-core/vcf_phase_shapeit5/main.nf

```diff
@@ -58,15 +57,15 @@ workflow VCF_PHASE_SHAPEIT5 {
     VCF_INDEX1(SHAPEIT5_PHASECOMMON.out.phased_variant)
     ch_versions = ch_versions.mix(VCF_INDEX1.out.versions.first())
 
-    ligate_input = SHAPEIT5_PHASECOMMON.output.phased_variant
-                                        .map{meta, vcf -> [meta, vcf]}
-                                        .groupTuple()
-                                        .combine(VCF_INDEX1.out.csi
-                                            .map{meta, vcf -> [meta, vcf]}
-                                            .groupTuple(),
-                                            by:0)
+    ch_ligate_input = SHAPEIT5_PHASECOMMON.output.phased_variant
+        .map{meta, vcf -> [groupKey(meta), vcf]}
```

Comment:
You should also add a size to the `groupKey` indicating how large the group with this meta should be. e.g:

```
        .map{meta, vcf, group_size -> [groupKey(meta, group_size), vcf]}
```

Where you defined before how many chunks there are for each group and stored that as the `group_size` value

Reply:
Ah okay !